models/agent.go

```diff
@@ -182,3 +184,79 @@ type RDSExporter struct {
 
 	ListenPort *uint16 `reform:"listen_port"`
 }
+
+// AgentFilters represents filters for agents list.
+type AgentFilters struct {
+	// Return only Agents running on that Node.
+	HostNodeID string
+	// Return only Agents that provide insights for that Node.
+	NodeID string
+	// Return only Agents that provide insights for that Service.
+	ServiceID string
+}
+
+// AgentsByFilters returns agents providing insights for a given filters sorted by Agent ID.
+func AgentsByFilters(q *reform.Querier, filters AgentFilters) ([]*AgentRow, error) {
+	var joins []string
+	var conditions []string
+	var args []interface{}
+	var structs []reform.Struct
+	var err error
+	filtered := false
+	if filters.HostNodeID != "" {
+		qi := q.QualifiedView(AgentRowTable) + "." + q.QuoteIdentifier("runs_on_node_id")
+		condition := fmt.Sprintf("%s = ?", qi)
+		conditions = append(conditions, condition)
+		args = append(args, filters.HostNodeID)
+		filtered = true
+	}
+	if filters.NodeID != "" {
+		query := fmt.Sprintf("inner join %s on %s.%s = %s.%s",
+			q.QualifiedView(AgentNodeView),
+			q.QualifiedView(AgentRowTable), q.QuoteIdentifier("id"),
+			q.QualifiedView(AgentNodeView), q.QuoteIdentifier("agent_id"),
+		)
+		joins = append(joins, query)
+
+		qi := q.QualifiedView(AgentNodeView) + "." + q.QuoteIdentifier("node_id")
+		condition := fmt.Sprintf("%s = ?", qi)
+		conditions = append(conditions, condition)
+		args = append(args, filters.NodeID)
+		filtered = true
+	}
+	if filters.ServiceID != "" {
+		query := fmt.Sprintf("inner join %s on %s.%s = %s.%s",
+			q.QualifiedView(AgentServiceView),
+			q.QualifiedView(AgentRowTable), q.QuoteIdentifier("id"),
+			q.QualifiedView(AgentServiceView), q.QuoteIdentifier("agent_id"),
```

Comment:
`QualifiedView` and `QuoteIdentifier` should be used to sanitize user's input. In that case, it is just noise. Nothing is wrong with
```go
query := `INNER JOIN agent_services ON agents.id = agent_services.agent_id
```